syntax = "proto3";

service Game {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc GetState (GetStateRequest) returns (StateResponse) {}
  rpc UpdateLocation (UpdateLocationRequest) returns (StateResponse) {}
}

// The request message containing the user's name.
message HelloRequest {
  int32 client_version = 1;
}

// The response message containing the greetings
message HelloReply {
  bool valid_client_version = 1;
  int32 server_version = 2;
}

message UpdateLocationRequest {
    float x = 1;
    float y = 2;
}

message GetStateRequest {
}

message StateResponse {
    float location_x = 1;
    float location_y = 2;
}
